app/components/qr/image-preview.tsx

```diff
@@ -0,0 +1,131 @@
+import {
+  useEffect,
+  useImperativeHandle,
+  useRef,
+  useState,
+  forwardRef,
+} from "react";
+import { loadImage } from "~/utils/load-image";
+import type {
+  ImagePreviewProps,
+  ImagePreviewState,
+  ImagePreviewRef,
+} from "./types";
+
+// NOTE: Do not trim the space at the end.
+// It used for spacing between the logo and the text.
+const POWERED_BY = "Powered by  ";
+const FONT_SIZE_MAP = {
+  cable: 0,
+  small: 8,
+  medium: 12,
+  large: 16,
+};
+const LOGO_SIZE_MAP = {
```

Comment:
hey @tobihans . The designers looked at this and we have to make another modification. We use those sizes for printing so we have to adjust the sizes a bit.
Here is what @jurrejansen shared:

```
QR sizes

Small = 20mm = 56,6929px
Font size = 3pt = 4px
Icon = 4,9606px

Medium = 30mm = 80,7874px
Font size = 4pt = 5px
Icon = 7,4409px

Large = 40mm = 107,7165
Font size = 6pt = 8px
Icon = 9,9213px
```

Designs:
![image](https://github.com/Shelf-nu/shelf.nu/assets/7840007/500a1a0f-5605-4a4b-bc69-e84cd65f2555)
![image](https://github.com/Shelf-nu/shelf.nu/assets/7840007/3e8cc91d-25da-481e-9df9-0ec7fe3576d7)
![image](https://github.com/Shelf-nu/shelf.nu/assets/7840007/a6c97c53-832e-4dbb-b062-7f3046c1c9c6)

@tobihans if anything is not clear, lets discuss it in discord with you and @jurrejansen so we can finalize this.

Reply:
@jurrejansen please mention to Tobi that, the vision is to be able to allow users to change the background of the codes in the future. Not on this release of course but it is good for him to know.

Reply:
@tobihans yes, future vision is to able to change the color of the border around the QR code (see screenshot) I would like the grey border to be the default (#F2F4F7). Could you also make sure the 'Powered by *icon*' is vertically aligned to center. (see screenshot) I used a design with orange border to make more clear.

![Screenshot 2023-08-22 at 14 28 23](https://github.com/Shelf-nu/shelf.nu/assets/76112272/e0b8c5ab-00b2-4a91-9003-4e0ecf785fce)

![Screenshot 2023-08-22 at 14 45 08](https://github.com/Shelf-nu/shelf.nu/assets/76112272/6b0e665b-8246-40d1-aad9-96393141a21c)

Reply:
To add here what we have written on discord. The sizes of the canvas itself should be as follows:
- 20 millimeter = 75.6px
- 30 millimeter = 113.38px
- 40 millimeter = 151.18px

Reply:
@jurrejansen I tried my best to vertically align but it's not pixel perfect to be honest.
In fact, technically, I don't have the measures to make the right calculations. The black cells grid and the white background form the original image, so I can't get distances programmatically to do what you said. 

Other than that, I believe everything is fine. Thanks